x/txfees/keeper/hooks.go

```diff
@@ -0,0 +1,64 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochstypes "github.com/osmosis-labs/osmosis/v7/x/epochs/types"
+	txfeestypes "github.com/osmosis-labs/osmosis/v7/x/txfees/types"
+)
+
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {}
+
+// at the end of each epoch, swap all non-OSMO fees into OSMO and transfer to fee module account
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
+	addrNonNativeFee := k.accountKeeper.GetModuleAddress(txfeestypes.NonNativeFeeCollectorName)
+	nonNativeFeeAccountBalances := k.bankKeeper.GetAllBalances(ctx, addrNonNativeFee)
+	baseDenom, _ := k.GetBaseDenom(ctx)
+
+	for _, coin := range nonNativeFeeAccountBalances {
+		if coin.Denom == baseDenom {
+			continue
+		} else {
```

Comment:
there's no need to wrap this in an `else` block as the `if` condition, if matched, will just continue. I'm surprised the linter didn't signal this.

Reply:
Great point - should be fixed in next commit.